moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -572,16 +573,27 @@ bool PlanningSceneMonitor::requestPlanningSceneState(const std::string& service_
   return false;
 }
 
-void PlanningSceneMonitor::providePlanningSceneService(const std::string& service_name)
+void PlanningSceneMonitor::providePlanningSceneService(const std::string& apply_service_name, const std::string& get_service_name)
 {
   // Load the service
+  apply_scene_service_ = pnode_->create_service<moveit_msgs::srv::ApplyPlanningScene>(
+      apply_service_name, [this](moveit_msgs::srv::ApplyPlanningScene::Request::SharedPtr req,
+                           moveit_msgs::srv::ApplyPlanningScene::Response::SharedPtr res) { return applyPlanningSceneServiceCallback(req, res); });
+
   get_scene_service_ = pnode_->create_service<moveit_msgs::srv::GetPlanningScene>(
-      service_name, [this](moveit_msgs::srv::GetPlanningScene::Request::SharedPtr req,
+      get_service_name, [this](moveit_msgs::srv::GetPlanningScene::Request::SharedPtr req,
                            moveit_msgs::srv::GetPlanningScene::Response::SharedPtr res) {
         return getPlanningSceneServiceCallback(req, res);
       });
 }
 
+void PlanningSceneMonitor::applyPlanningSceneServiceCallback(std::shared_ptr<moveit_msgs::srv::ApplyPlanningScene::Request> req,
+                                           std::shared_ptr<moveit_msgs::srv::ApplyPlanningScene::Response> res)
+{
+  updateFrameTransforms();
```

Comment:
It shouldn't be necessary to update the transforms. Also, it's important to notice that `newPlanningSceneMessage()` only slots a new update event, the actual change will be applied after this service call has returned successfully (and might still fail).  If you want to apply the planning scene immediately, you'd have to call `PlanningScene::usePlanningSceneMsg()` while locking the scene.

Reply:
I do the above in the following section: https://github.com/peterdavidfagan/moveit2/blob/642680ec7fb9d16e470d314b1a50201b1b967b65/moveit_py/src/moveit_py/moveit_ros/planning_scene_monitor/planning_scene_monitor.cpp#L96. 

I am not entirely sure why I made this update, I am considering removing it.

Reply:
@henningkayser should I remove the apply planning scene service altogether? I think this was only leveraged by the service was only leveraged by the planning_scene_interface package which we no longer include in the bindings.

Reply:
[790223c](https://github.com/ros-planning/moveit2/pull/1546/commits/790223cfbcb14277805c755f1568f71c9c2e66ae)